en/concept/flight_modes.md

```diff
@@ -1,111 +1,215 @@
-# Flight Modes (Developers) 
+# Flight Modes (Developers)
 
-_Flight Modes_ define how the autopilot responds to user input and controls vehicle movement.
+Flight modes are special operational states that control how the autopilot responds to user input and controls vehicle movement.
 They are loosely grouped into _manual_, _assisted_ and _auto_ modes, based on the level/type of control provided by the autopilot.
 The pilot transitions between flight modes using switches on the remote control or with a ground control station.
 
-Not all flight modes are available on all vehicle types, and some modes behave differently on different vehicle types (as described below).
-Finally, some flight modes make sense only under specific pre-flight and in-flight conditions (e.g. GPS lock).
-The system will not allow transitions to those modes until the right conditions are met.
+Flight modes can be implemented in the PX4 flight stack running on the flight controller, or as PX4 ROS 2 modes running on a companion computer.
+From an external perspective (MAVLink/GCS), the origin of a mode is indistinguishable.
 
-The sections below provide an overview of the modes, followed by a [flight mode evaluation diagram](#flight-mode-evaluation-diagram) that shows the conditions under which PX4 will transition into a new mode.
+This topic links to documentation for the supported modes, compares PX4 FC and ROS2 modes, provides implementation hints, and provides an overview of PX4 modes can be used with MAVLink.
 
-::: info
-User-facing flight mode documentation can be found in:
+## Supported Flight Modes
+
+Not all flight modes are available (or makes sense), on all vehicle types, and some modes behave differently on different vehicle types.
+
+Flight mode documentation for the modes running on the flight controller are listed below:
 
 - [Flight Modes (Multicopter)](../flight_modes_mc/index.md)
 - [Flight Modes (Fixed-Wing)](../flight_modes_fw/index.md)
 - [Flight Modes (VTOL)](../flight_modes_vtol/index.md)
 - [Drive Modes (Differential Rover)](../flight_modes_rover/differential.md)
 - [Drive Modes (Ackermann Rover)](../flight_modes_rover/ackermann.md)
+- [Basic Configuration > Flight Modes](../config/flight_mode.md)
+
+::: info
+At time of writing there are no PX4 ROS 2 flight modes that are considered "supported as an integral part of the PX4 open source offering".
+:::
+
+## PX4 FC Modes vs PX4 ROS2 Flight Modes
```

Comment:
FYI, this section put me in contortions - how do I refer to a flight mode that runs in the FC?

It was the reason I wrote this https://github.com/PX4/PX4-user_guide/issues/3565#issuecomment-2655407569

Reply:
Should we label them with "internal" (vs. "external"=running on ROS2)

Reply:
I like Internal vs External @sfuhrer

Reply:
@mrpollo @sfuhrer I "like" Internal vs External too, but I'm not certain it is future proof terminology. 

Assuming you read https://github.com/PX4/PX4-user_guide/issues/3565#issuecomment-2655407569, are you suggesting that if in future we have a flight stack that spans FC and Companion, then we will refer to the parts as _PX4 Internal_ (runs on FC) and _PX4 External_ (runs on companion)?

It's a little clunky in that it will make any code on the companion a second class citizen. That might be OK. 
I proposed PX4 RTOS modes / PX4 GTOS modes, but if you are both happy with internal external for this, we can give it a go.

Reply:
For now I will try Internal vs External, but I would like you to think about ^^^^

Reply:
> "Internal" implies "you've got to be on the flight stack to really be part of PX4".

But that's how it is - we (PX4 community) do not maintain the ROS flight modes, so the user is on it's own when they use them.

I btw fully support you in spending the time now to think this through - it will be very hard to change in the future once people get used to the naming.

Reply:
Yes, that is exactly the problem.

If I knew we'd have "full stack" PX4 in a year or so I'd push for `PX4 RTOS modes` and `PX4 GCOS modes` (or replacing RTOS/GCOS with RT, NuttX, GC, Linux).